db/crud.go

```diff
@@ -550,7 +550,11 @@ func (col *DatabaseCollectionWithUser) authorizeDoc(doc *Document, revid string)
 	if revid == "" {
 		revid = doc.CurrentRev
 	}
-	if rev := doc.History[revid]; rev != nil {
+
+	if revid == doc.CurrentRev {
+		ch := doc.currentChannels()
+		return col.user.AuthorizeAnyCollectionChannel(col.ScopeName, col.Name, ch)
```

Comment:
This is sub-optimal - `currentChannels` iterates over `doc.SyncData.Channels` to determine the active set of channels for an active revision.